src/main/java/liquibase/ext/neo4j/lockservice/Neo4jLockService.java

```diff
@@ -0,0 +1,124 @@
+package liquibase.ext.neo4j.lockservice;
+
+import liquibase.Scope;
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LiquibaseException;
+import liquibase.exception.LockException;
+import liquibase.executor.Executor;
+import liquibase.executor.ExecutorService;
+import liquibase.ext.neo4j.database.Neo4jDatabase;
+import liquibase.lockservice.DatabaseChangeLogLock;
+import liquibase.lockservice.LockService;
+import liquibase.statement.SqlStatement;
+import liquibase.statement.core.RawSqlStatement;
+
+import java.util.Collections;
+import java.util.UUID;
+
+public class Neo4jLockService implements LockService {
+
+    private static final String CONSTRAINT_NAME = "unique_liquibase_lock";
+    // this query will only work with Neo4j 4.x (not 3.x, as db.constraints() does not yield the 'name' column)
+    private static final String CHECK_LOCK_CONSTRAINT_QUERY = String.format("CALL db.constraints() YIELD name\n" +
+            "WITH collect(name) AS names\n" +
+            "RETURN NONE(name IN names WHERE name = \"%1$s\" OR name = \"%1$s_bc\") AS result", CONSTRAINT_NAME);
+
+    private Database database;
+
+    @Override
+    public int getPriority() {
+        return PRIORITY_DATABASE;
+    }
+
+    @Override
+    public boolean supports(Database database) {
+        return database instanceof Neo4jDatabase;
+    }
+
+    @Override
+    public void setDatabase(Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public void setChangeLogLockWaitTime(long changeLogLockWaitTime) {
+
+    }
+
+    @Override
+    public void setChangeLogLockRecheckTime(long changeLogLocRecheckTime) {
+
+    }
+
+    @Override
+    public boolean hasChangeLogLock() {
+        return true;
+    }
+
+    @Override
+    public void waitForLock() throws LockException {
+
+    }
+
+    @Override
+    public void init() throws DatabaseException {
```

Comment:
@nvoxland I've checked the Liquibase implementation but I don't 100% follow where `init` should be called.

It seems to be called in `Liquibase#checkLiquibaseTables` but also in `StandardLockService#acquireLock` and `StandardLockService#releaseLock`.
Isn't `init` already called before lock acquire/release happens?

Reply:
It needs a bit of cleanup (I've started on it in a branch). There are a few places where init() is called prior to aquireLock/releaseLock, but there are enough paths to it (especially in direct API usage), that it really needs to be up to the lock service to make sure it is ready whenever aquireLock and releaseLock is called.

So you will want to call init() yourself as well, and make sure you are OK with it being called multiple times.

Reply:
Quick follow-up question: in my case, `init` only sets up uniqueness constraints (there is no such thing as a table in Neo4j). I assume not calling `init` in `releaseLock` is fine in that case? (I don't need the uniqueness constraint to be set to delete a node in the graph)

Reply:
Yes, that should be fine I think